src/activities/morse_code/MorseCode.qml

```diff
@@ -0,0 +1,573 @@
+            property alias score: score
+            property alias textInput: textInput
+            property var dataset: Dataset.get()
+            property bool toAlpha: dataset[currentLevel].toAlpha
+            property string questionText: dataset[currentLevel].question
+            property string questionValue
+            property string instruction: dataset[currentLevel].instruction
+            property int currentLevel: 0
+            property int numberOfLevel: dataset.length
+
+            onToAlphaChanged: {
+                textInput.text = ''
+                morseConverter.alpha=0
+                if(toAlpha)
+                    keyboard.populateAlpha()
+                else
+                    keyboard.populateMorse()
+            }
+
+            function start() {
+                if (!ApplicationInfo.isMobile)
+                    textInput.forceActiveFocus();
+                items.currentLevel = 0
+                initLevel()
+            }
+
+            function initLevel() {
+                score.currentSubLevel = 1
+                score.numberOfSubLevels = dataset[currentLevel].values[1].length
+                initSubLevel()
+            }
+
+            function initSubLevel() {
+                if(dataset[currentLevel].values[0] == '_random_') {
+                    var randomIndex = Math.floor(Math.random() * dataset[currentLevel].values[1].length)
+                    questionValue = dataset[currentLevel].values[1][randomIndex]
+                    dataset[currentLevel].values[1].splice(randomIndex,1)
+                } else {
+                    questionValue = dataset[currentLevel].values[1][score.currentSubLevel - 1]
+                }
+            }
+
+            function nextLevel() {
+                if(numberOfLevel - 1 == currentLevel ) {
+                    currentLevel = 0
+                } else {
+                    currentLevel++
+                }
+
+                initLevel();
+            }
+
+            function previousLevel() {
+                if(currentLevel == 0) {
+                    currentLevel = numberOfLevel - 1
+                } else {
+                    currentLevel--
+                }
+
+                initLevel();
+            }
+
+            function nextSubLevel() {
+                if(++score.currentSubLevel > score.numberOfSubLevels) {
+                    nextLevel()
+                } else {
+                    initSubLevel();
+                }
+            }
+
+        }
+
+        onStart: {
+            first_screen.visible=true
+            items.start()
+        }
+        onStop: {}
+
+        QtObject {
+            id: morseConverter
+            property string alpha
+            property string morse
+            property string last
+
+            function morse2alpha(str) {
+                var table =
+                {
+                 ".-" : "A" , "-...":"B","-.-.":"C","-..":"D",".":"E","..-.":"F","--.":"G",
```

Comment:
why not make two arrays one with morse code and other for characters with corresponding index.